onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java

```diff
@@ -1,26 +1,81 @@
 package com.onfido.integration;
 
 import java.io.IOException;
+import com.onfido.Onfido;
+import com.onfido.Onfido.Builder;
+import com.onfido.JsonObject;
+
+import static org.junit.Assert.assertEquals;
+
 import okhttp3.mockwebserver.MockResponse;
 import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
 import org.testng.annotations.AfterTest;
 import org.testng.annotations.BeforeTest;
 
 class ApiIntegrationTest {
+
+  private String apiToken = System.getenv("ONFIDO_API_TOKEN");
+  private Onfido.Builder builder = Onfido.builder();
+  protected Onfido onfido;
+
+  private RecordedRequest request;
   private MockWebServer server;
 
-  @BeforeTest
-  public void clearServer() {
-    server = null;
+  protected ApiIntegrationTest() {
+    if ( ! mockingEnabled() )
+    {
+      onfido = builder.apiToken(apiToken).regionEU().build();
+    }
   }
 
+  @BeforeTest
+  private void setup() {}
+
   @AfterTest
-  public void shutdownServer() throws IOException {
-    if (server != null) {
+  private void tearDown() throws IOException {
+    if ( mockingEnabled() && server != null )
+    {
       server.shutdown();
     }
   }
 
+  protected boolean mockingEnabled()
+  {
+    return ( apiToken == null );
+  }
+
+  public void prepareMock(JsonObject mockedResponse) throws IOException
+  {
+    if ( mockingEnabled() )
+    {
+      server = new MockWebServer();
+      server.enqueue(new MockResponse().setBody(mockedResponse.toJson()));
+      server.start();
+
+      onfido = builder.apiToken("token").unknownApiUrl(server.url("/").toString()).build();
+    }
+  }
```

Comment:
Code style to be improved here too. As well as on the remaining of the whole file 💄

Reply:
Are you just referring to:
- newline before closing curly bracket
- spaces in-between round brackets
Or is there any other code style improvement you're suggesting?

Reply:
> * Code style could be improved a bit to a more java'ish approach
> * Some help methods are floating around test files. I'd suggest that we extract those into helpers if you agree 💡

Code style should be better now. Regarding the helper class, if you're referring to new methods inside `ApiIntegrationTest` class, that was the case at the beginning but it was making code in the test classes less readable (due to the call to the helper). Moreover in my understanding the aim `ApiIntegrationTest` is really to perform as an helper (all methods in the class looks fulfilling that role and the 3 last methods are expected to disappear as replaced by the new ones).